src/components/Form.js

```diff
@@ -17,32 +16,46 @@ export default class Form extends Component {
     noValidate: false,
     liveValidate: false,
     safeRenderCompletion: false,
-  }
+  };
 
   constructor(props) {
     super(props);
     this.state = this.getStateFromProps(props);
   }
 
+  componentDidMount() {
+    const {formData} = this.state;
+    const {liveValidate, noValidate, schema} = this.props;
+    const {errors = [], errorSchema = {}} = this.state;
+    const mustValidate = !!formData && !noValidate && liveValidate;
+
+    Promise
```

Comment:
Well

1. they're [coming](http://caniuse.com/#feat=async-functions)
1. they're transpiled by Babel so we don't really care with platform implementations
1. they improves by several orders of magnitude the readability of async flows
1. part of our job is to challenge upcoming standards

We've switched to async/await in Kinto.js [lately](https://github.com/Kinto/kinto.js/pull/548/files) and there's simply no way back. The Promise API is verbose, hard to read and quite cumbersome when it comes to handling errors properly.